modules/platforms/dotnet/Apache.Ignite.Linq/Impl/MethodVisitor.cs

```diff
@@ -329,6 +331,50 @@ private static void AppendAdjustment(CacheQueryExpressionVisitor visitor, int[]
             visitor.Parameters.Add(paramValue);
         }
 
+        /// <summary>
+        /// Get IgnoreCase parameter for string.Compare method
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        private static bool GetStringCompareIgnoreCaseParameter(Expression expression)
+        {
+            if (expression is ConstantExpression constant)
+            {
+                if (constant.Value is bool ignoreCase)
+                {
+                    return ignoreCase;
+                }
+            }
+
+            throw new ArgumentException("ignoreCase");            
+        }
+
+        /// <summary>
+        /// Visits string.Compare method
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <param name="visitor"></param>
+        private static void VisitStringCompare(MethodCallExpression expression, CacheQueryExpressionVisitor visitor, bool ignoreCase)
+        {
+            visitor.ResultBuilder.Append("casewhen(");
+            if (ignoreCase) visitor.ResultBuilder.Append("lower(");
+            visitor.Visit(expression.Arguments[0]);
+            if (ignoreCase) visitor.ResultBuilder.Append(")");
+            visitor.ResultBuilder.Append(" = ");
+            if (ignoreCase) visitor.ResultBuilder.Append("lower(");
+            visitor.Visit(expression.Arguments[1]);
+            if (ignoreCase) visitor.ResultBuilder.Append(")");
+            visitor.ResultBuilder.Append(", 0, casewhen(");
+            visitor.ResultBuilder.Append("lower(");
```

Comment:
It seems that string.Compare method considers case only for string equality and ignores case while defining if string is greater than the other. Looks strange for me too. 
You can run these lines of code:
``		
Console.WriteLine(string.Compare("Person_1300", "Person_1300", false)); // 0
Console.WriteLine(string.Compare("Person_1299", "Person_1300", false)); // -1
Console.WriteLine(string.Compare("Person_1301", "Person_1300", false)); // 1
Console.WriteLine(string.Compare("Person_1300", "person_1300", false)); // 1
Console.WriteLine(string.Compare("Person_1299", "person_1300", false)); // -1
Console.WriteLine(string.Compare("Person_1301", "person_1300", false)); // 1
``
On the opposite side ignite considers case for string comparison:
``
SELECT 'Person_1300' < 'person_1300', 'Person_1299' < 'person_1300', 'Person_1301' < 'person_1300'; // true, true, true
``
Thus to mimic .NET method behavior I had to always convert strings to lower case while defining if string is greater than the other.

Reply:
The difference is caused by the fact that this `string.Compare` overload is culture-specific (uses `CultureInfo.CurrentCulture`).

I don't think we should try to mimic this behavior when translating to SQL.
Let's keep it simple, [like EF Core does](https://docs.microsoft.com/en-us/ef/core/miscellaneous/collations-and-case-sensitivity#translation-of-built-in-net-string-operations for equality `EF Core makes no attempt to translate simple equality to a database case-sensitive operation`.

* If `ignoreCase` is set - use `lower`, otherwise don't.
* Alternatively, drop the support for this overload to avoid confusion, keep only `Compare(string, string)`

Reply:
@ptupitsyn, I did what was easiest: updated VisitStringCompare method to use ignoreCase in both cases + updated unit tests